scripts/ci/gitlab/pipeline/publish.yml

```diff
@@ -233,11 +233,19 @@ publish-draft-release:
 
 publish-crates:
   extends:                         .publish-crates-template
+  # publish-crates should only be run if publish-crates-locally passes
+  needs:
+    - job:                         publish-crates-locally
+      artifacts:                   false
 
 publish-crates-manual:
   extends:                         .publish-crates-template
   when:                            manual
   interruptible:                   false
+  # publish-crates should only be run if publish-crates-locally passes
```

Comment:
This dependency is not necessary for the manual version. The manual version is supposed to be triggered ad-hoc, only if necessary, without any requirements. In fact, could you remove `stage: publish` from `.publish-crates-template` and move it to `publish-crates`?

Reply:
Okay, I remove dependency, but I didn't understand about the stage. Each job must have a stage and if I do that the `publish-crates-manual` job will be without stage.

Reply:
> Okay, I remove dependency, but I didn't understand about the stage. Each job must have a stage and if I do that the `publish-crates-manual` job will be without stage.

Ideally that job would appear at the start of the pipeline, like the [weights jobs](https://github.com/paritytech/polkadot/blob/4ea6ed09c0a605bff3a2f0216cc3ab4f5c60d6cd/scripts/ci/gitlab/pipeline/weights.yml#L10) which appear at [the start of the Polkadot pipeline](https://github.com/paritytech/polkadot/blob/4ea6ed09c0a605bff3a2f0216cc3ab4f5c60d6cd/.gitlab-ci.yml#L18), such that it'll be possible to reliably trigger it without the risk of it being cancelled due to it being positioned in the later stages. I guess `stage: .pre` would work?

Anyways, if that idea sounds unreasonable, I don't mind leaving the stage as it is. Please just remove the requirement, then.